src/parser_urdf.cc

```diff
@@ -2663,48 +2665,137 @@ void URDF2SDF::ListSDFExtensions(const std::string &_reference)
-             << static_cast<int>(_link->child_links.size())
-             << "] children links ignored.\n";
+      errorStream << "a mass value of less than or equal to zero. ";
     }
+    errorStream << "Please ensure this link has a valid mass to prevent any "
+                << "missing links or joints in the resulting SDF model.";
+    nonJointReductionErrors.emplace_back(
+        ErrorCode::LINK_INERTIA_INVALID, errorStream.str());
+    errorStream.str(std::string());
 
-    if (!_link->child_joints.empty())
+    // If joint reduction happens and resolves the issue of this link with no
+    // mass, no warnings will be emitted
+    bool jointReductionHappens = _link->parent_joint &&
+        FixedJointShouldBeReduced(_link->parent_joint) &&
+        g_reduceFixedJoints;
+
+    // check if parent joint can be lumped and reduced
+    if (!jointReductionHappens && _link->parent_joint)
     {
-      sdfdbg << "urdf2sdf: link[" << _link->name
-             << "] has no inertia, ["
-             << static_cast<int>(_link->child_links.size())
-             << "] children joints ignored.\n";
+      if (_link->parent_joint->type == urdf::Joint::FIXED &&
+          (!FixedJointShouldBeReduced(_link->parent_joint) ||
+              !g_reduceFixedJoints))
+      {
+        errorStream << "urdf2sdf: allowing joint lumping by removing any "
+                    << "<disableFixedJointLumping> or <preserveFixedJoint> "
+                    << "gazebo tag on fixed parent joint["
+                    << _link->parent_joint->name
+                    << "], as well as ensuring that "
+                    << "ParserConfig::URDFPreserveFixedJoint is false, could "
+                    << "help resolve this warning. See [URL_HERE] for more "
+                    << "information about this behavior.";
+        nonJointReductionErrors.emplace_back(
+            ErrorCode::LINK_INERTIA_INVALID, errorStream.str());
+        errorStream.str(std::string());
+      }
+
+      errorStream << "urdf2sdf: parent joint["
+                  << _link->parent_joint->name
+                  << "] ignored.";
+      nonJointReductionErrors.emplace_back(
+          ErrorCode::LINK_INERTIA_INVALID, errorStream.str());
+      errorStream.str(std::string());
     }
 
-    if (_link->parent_joint)
+    // check child joints for any possibility of joint lumping and reduction
+    if (!jointReductionHappens && !_link->child_joints.empty())
     {
-      sdfdbg << "urdf2sdf: link[" << _link->name
-             << "] has no inertia, "
-             << "parent joint [" << _link->parent_joint->name
-             << "] ignored.\n";
+      for (const auto &cj : _link->child_joints)
+      {
+        if (FixedJointShouldBeReduced(cj) && g_reduceFixedJoints)
+        {
+          jointReductionHappens = true;
+          break;
+        }
+        else if (cj->type == urdf::Joint::FIXED &&
+            (!FixedJointShouldBeReduced(cj) || !g_reduceFixedJoints))
```

Comment:
Can this just be an `else`? The joint type is checked in `FixedJointShouldBeReduced`.

Reply:
We will still need to check that the child joint is fixed at least, as `FixedJointShouldBeReduced` returning false could be due to gazebo tags.

The 2nd line statement has been cleaned up, https://github.com/gazebosim/sdformat/pull/1238/commits/bb9fa6367488c3fac01be992f2347e4998f404d2